script/tool/lib/src/gradle_check_command.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+
+import 'common/core.dart';
+import 'common/package_looping_command.dart';
+import 'common/repository_package.dart';
+
+/// A command to enforce gradle file conventions and best practices.
```

Comment:
This is not true based on the implementation. I think this will only impact topline gradle files and does not check gradle files in app. 

Also from a structural perspective I think you might want to logically separate example gradle files from gradle files of plugins that are published.

Reply:
> I think this will only impact topline gradle files and does not check gradle files in app.

This is currently true, because the check I've added didn't seem important for apps. If that's not the case we can definitely change that.

> Also from a structural perspective I think you might want to logically separate example gradle files from gradle files of plugins that are published.  

Yes, that's why I have it only iterating over top-level packages, rather than setting the command to iterate over everything. My thought was that we could add example checks later, but I can definitely implement the structure for that now.

Reply:
It now iterates over everything, and just skips the current check for examples, so it will be easy to add things for all gradle files later.

(Let me know if you want this specific check to run for examples; currently it's not even in the `flutter` template for apps.)